posthog/hogql/database/schema/util/test/test_session_where_clause_extractor.py

```diff
@@ -344,7 +344,7 @@ def test_select_with_timestamp(self):
     FROM
         sessions
     WHERE
-        and(equals(sessions.team_id, <TEAM_ID>), greaterOrEquals(plus(toTimeZone(sessions.min_timestamp, %(hogql_val_1)s), toIntervalDay(3)), %(hogql_val_2)s))
+        and(equals(sessions.team_id, <TEAM_ID>), ifNull(greaterOrEquals(toTimeZone(sessions.min_timestamp, %(hogql_val_1)s), minus(%(hogql_val_2)s, toIntervalDay(3))), 0))
```

Comment:
Need to track this new `ifNull` (and others below) down as it will likely be a performance regression otherwise

Reply:
I wrote this in slack:

I'm just looking at the hogql printer.py, it looks like we have a special case for when we print a comparison operation on a timestamp field https://github.com/PostHog/posthog/blob/5bea2bdfc053ca1aee3be4d93c759152676db79c/posthog/hogql/printer.py#L794

We could add another special case for this comparison here

Reply:
Piggybacking off of #30628 now